src/common/translation.cpp

```diff
@@ -125,6 +125,54 @@ wxString GetPreferredUILanguageFallback(const wxArrayString& WXUNUSED(available)
     return lang;
 }
 
+// When the preferred UI language is determined, the LANGUAGE environment
+// variable is the primary source of preference.
+// http://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html
+//
+// The LANGUAGE variable may contain a colon separated list of language
+// codes in the order of preference.
+// http://www.gnu.org/software/gettext/manual/html_node/The-LANGUAGE-variable.html
+wxString GetPreferredUILanguageFromEnvVariable(const wxArrayString& available)
+{
+    wxString languageFromEnv;
+    wxArrayString preferred;
+    if ( wxGetEnv("LANGUAGE", &languageFromEnv) )
+    {
+        wxStringTokenizer tknzr(languageFromEnv, ":");
+        while ( tknzr.HasMoreTokens() )
+        {
+            const wxString tok = tknzr.GetNextToken();
+            if ( const wxLanguageInfo *li = wxLocale::FindLanguageInfo(tok) )
+            {
+                preferred.push_back(li->CanonicalName);
+            }
+        }
+        if ( preferred.empty() )
+        {
+            wxLogTrace(TRACE_I18N, " - LANGUAGE was set, but it didn't contain existing languages");
```

Comment:
The wording here is a bit weird, I think "recognized" would be more clean than "existing" (if I understand correctly what is this trying to say).

Reply:
I'll rephrase it as "... didn't contain any languages recognized by the system" or something like that.